tests/components/webmin/test_config_flow.py

```diff
@@ -0,0 +1,108 @@
+"""Test the Webmin config flow."""
+from __future__ import annotations
+
+from http import HTTPStatus
+from typing import Any
+from unittest.mock import AsyncMock, patch
+from xmlrpc.client import Fault
+
+from aiohttp.client_exceptions import ClientConnectionError, ClientResponseError
+import pytest
+
+from homeassistant import config_entries
+from homeassistant.components.webmin.const import DOMAIN
+from homeassistant.const import CONF_HOST, CONF_NAME
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import FlowResultType
+
+from .conftest import TEST_USER_INPUT_FULL, TEST_USER_INPUT_REQUIRED
+
+from tests.common import load_json_object_fixture
+
+pytestmark = pytest.mark.usefixtures("mock_setup_entry")
+
+
+@pytest.fixture
+async def user_flow(hass):
+    """Return a user-initiated flow after filling in host info."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+    assert result["type"] == FlowResultType.FORM
+    assert result["errors"] is None
+    return result["flow_id"]
+
+
+async def test_show_form_user(hass: HomeAssistant) -> None:
+    """Test showing the form to select the authentication type."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+    assert result["type"] == FlowResultType.FORM
+    assert result["step_id"] == "user"
+    assert result["errors"] is None
+
+
+@pytest.mark.parametrize(
+    ("test_config", "expected_title"),
+    [
+        (TEST_USER_INPUT_FULL, TEST_USER_INPUT_FULL[CONF_NAME]),
+        (TEST_USER_INPUT_REQUIRED, TEST_USER_INPUT_REQUIRED[CONF_HOST]),
+    ],
+)
+async def test_form_user(
+    hass: HomeAssistant,
+    user_flow,
+    test_config: dict[str, Any],
+    expected_title: str,
+    mock_setup_entry: AsyncMock,
+):
+    """Test a successful user initiated flow."""
+    with patch(
+        "webmin_xmlrpc.client.WebminInstance.update",
+        return_value=load_json_object_fixture("webmin_update.json", DOMAIN),
+    ):
```

Comment:
Please patch the package where used and not globally.

Reply:
I don't quite understand what you mean with "where used and not globally". It is needed for the async_configure because of the validation.

Reply:
Sorry for not be specific here, hope the code explains it :)

```suggestion
    with patch(
        "homeassistant.components.webmin.config_flow.WebminInstance.update",
        return_value=load_json_object_fixture("webmin_update.json", DOMAIN),
    ):
```

Reply:
Ah, okay, I didn't know that you can patch functions on the place where they are imported :sweat_smile: